apps/generator/lib/conditionalGeneration.js

```diff
@@ -0,0 +1,267 @@
+const fs = require('fs').promises;
+const path = require('path');
+const log = require('loglevel');
+const logMessage = require('./logMessages');
+const jmespath = require('jmespath');
+
+/**
+ * Determines whether the generation of a file or folder should be skipped
+ * based on conditions defined in the template configuration.
+ *
+ * @param {string} relativeSourceFile - The relative path of the source file.
+ * @param {string} relativeSourceDirectory - The relative path of the source directory.
+ * @param {string} relativeTargetFile - The relative path of the target file to be generated.
+ * @param {Object} templateConfig - The template configuration containing conditional logic.
+ * @param {string} targetDir - The directory where the generated files are written.
+ * @param {string} matchedConditionPath - The matched path used to find applicable conditions.
+ * @param {Object} templateParams - Parameters passed to the template.
+ * @param {AsyncAPIDocument} asyncapiDocument - The AsyncAPI document used for evaluating conditions.
+ * @returns {Promise<boolean>} A promise that resolves to `true` if generation should be skipped, otherwise `false`.
+ */
+async function conditionalGeneration (
+  relativeSourceFile,
+  relativeSourceDirectory,
+  relativeTargetFile,
+  templateConfig,
+  targetDir,
+  matchedConditionPath,
+  templateParams,
+  asyncapiDocument 
+) {
+  const conditionFilesGeneration = templateConfig?.conditionalFiles?.[matchedConditionPath] || {};
+  const conditionalGeneration = templateConfig?.conditionalGeneration?.[matchedConditionPath] || {};
+
+  const config = Object.keys(conditionFilesGeneration).length > 0 
+    ? conditionFilesGeneration 
+    : conditionalGeneration;
+  
+  const parameter = config?.parameter;
+  const subject = config?.subject;
+  const validation = config?.validation;
+
+  if (Object.keys(conditionFilesGeneration).length > 0 && subject) {
+    return conditionalFilesGenerationDeprecatedVersion(
+      asyncapiDocument,
+      templateParams,
+      templateConfig,
+      relativeSourceFile
+    );
+  } else if (Object.keys(conditionalGeneration).length > 0 && subject) {
+    return conditionalSubjectGeneration(
+      asyncapiDocument,
+      templateConfig,
+      matchedConditionPath
+    );
+  }
```

Comment:
we need some comments in code with some more details as even I don't know why after this code we still have some more code. I mean, the condition is clear, if you have `conditionalFiles` in template - process, or if you have `conditionalGeneration` then process them.

now, if non of above is met, why we still do  parameters check and validation 🤔  some bacis parameter validation is already in `templateConfigValidator.js`

so yeah, I'm probably missing some context

Reply:
Noted, moved to simple if and else condition all code thanks @derberg